### DISCUSSION TITLE
Is there extension for C/C++ struct/class memory size and alignment?

### DISCUSSION THREAD

**Discussion by DolphinIQ** [microsoft/vscode-discussions#2434] [Extension Development QnA] [answered]
Visual Studio has this nice [feature](https://devblogs.microsoft.com/visualstudio/size-alignment-and-memory-layout-insights-for-c-classes-structs-and-unions/), which on struct/class hover shows a tooltip displaying it's memory size and alignment in bytes:

![image_2024-12-09_021019824](https://github.com/user-attachments/assets/f3a97ede-fd67-4540-8518-8fd913d1c60a)

This is not included in the VSCode official Microsoft Extension Pack. Does an extension for such functionality exist anywhere? I cant seem to find it and I like it a lot 🫤 . If a lot of people are interested, I could consider making such an extension

**Comment by alefragnani** [comment, marked as answer]
Hi @DolphinIQ ,

It's been a long time I don't play with C/C++, and after a quick look in the marketplace, I couldn't find any extension that provides what you asked for. I would suggest you to open a feature request in the Microsoft C/C++ extension, so the team could give you proper guidance, and I just noticed you already did that (https://github.com/microsoft/vscode-cpptools/issues/13034). 

It seems the team also use a upvote based approach for features, which seems great for your request, as you already have double digits votes. Also, as you commented that based on the number of people interested you could make the extension yourself, maybe, as an alternative, you could contribute a PR to the C/C++ extension, and take advantage of the team helping you in the process. And, if you have any questions in the future about making a new extension/contributing a PR, feel free to ask for help here too.

Hope this helps
